scylla-cdc/src/log_reader.rs

```diff
@@ -22,38 +29,8 @@ pub struct CDCLogReader {
 
 impl CDCLogReader {
     // Creates a reader and runs it immediately
-    #[allow(clippy::too_many_arguments)]
-    pub fn new(
-        session: Arc<Session>,
-        keyspace: String,
-        table_name: String,
-        start_timestamp: chrono::Duration,
-        end_timestamp: chrono::Duration,
-        window_size: time::Duration,
-        safety_interval: time::Duration,
-        sleep_interval: time::Duration,
-        consumer_factory: Arc<dyn ConsumerFactory>,
-    ) -> (Self, RemoteHandle<anyhow::Result<()>>) {
-        let (end_timestamp_sender, end_timestamp_receiver) =
-            tokio::sync::watch::channel(end_timestamp);
-        let mut worker = CDCReaderWorker::new(
-            session,
-            keyspace,
-            table_name,
-            start_timestamp,
-            end_timestamp,
-            window_size,
-            safety_interval,
-            sleep_interval,
-            end_timestamp_receiver,
-            consumer_factory,
-        );
-        let (fut, handle) = async move { worker.run().await }.remote_handle();
-        tokio::task::spawn(fut);
-        let printer = CDCLogReader {
-            end_timestamp: end_timestamp_sender,
-        };
-        (printer, handle)
+    pub fn new(end_timestamp: tokio::sync::watch::Sender<chrono::Duration>) -> Self {
```

Comment:
This method shouldn't be public anymore, it is useless for the users. Please send a followup which fixes the visibility (and the comment, as it is also incorrect).